lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -5441,7 +5441,15 @@ namespace Js
 
     Var JavascriptOperators::OP_InitCachedScope(Var varFunc, const Js::PropertyIdArray *propIds, Field(DynamicType*)* literalType, bool formalsAreLetDecls, ScriptContext *scriptContext)
     {
-        ScriptFunction *func = JavascriptGeneratorFunction::Is(varFunc) || JavascriptAsyncFunction::Is(varFunc) ?
+        bool isGAFunction = JavascriptFunction::Is(varFunc);
+        Assert(isGAFunction);
```

Comment:
Can you change the function signature to take a `JavascriptFunction *`? It looks like the 2 call sites are doing the right thing, but this is asking for trouble in the future